### PR TITLE
GH Actions fix

### PR DESCRIPTION
**Pull request by DenysGonchar**
This PR introduces the following changes:
- Replace of `ErlGang/setup-erlang@v1.0.0` with `erlef/setup-beam@v1` action + version update for other actions 
- Update of OTP versions used by GH Actions to `24.3.4` and `25.2`
- Update of GH Actions runners to `ubuntu-22.04`
- Split of the `big-tests` steps into a separate action
- Adding testing of dynamic domains profiles
- Re-enabling of `coveralls` reporting from GH Actions
- Fix of Jingle/SIP testing by GH Actions. codecov VS coveralls coverage report comparison showed that Jingle/SIP functionality previously was not tested by GH Actions at all.
- Elimination of coverage reporting for automatically generated modules (`XmppAddr` and `eldap_filter_yecc`):
   - This is required for aligning `coveralls` and `codecov` coverage reporting
   - `codecov` simply excludes missing in GH repo modules from the total coverage calculations, while `coveralls` includes them.
   - However, that doesn't make much sense since analysis of such unavailable files is impossible using `coveralls` service
   - As an example, see how [XmppAddr](https://coveralls.io/builds/58209545/source?filename=src%2FXmppAddr.erl) module is reported and compare it to [xml](https://coveralls.io/builds/58209545/source?filename=src%2Fxml.erl) module.
   - missing functionality has been added to rebar3_codecov plugin, check [this](https://github.com/esl/rebar3_codecov/commit/facb183def1643f70e4bb6fc4e35078d23f9caff) commit for more details.

[codecov.io report](https://app.codecov.io/gh/esl/MongooseIM/tree/GH-actions-fix)
[coveralls.io reports](https://coveralls.io/github/esl/MongooseIM?branch=GH-actions-fix)
[GH Actions reports](https://github.com/esl/MongooseIM/actions/workflows/ci.yml?query=branch%3AGH-actions-fix)
[CircleCI reports](https://app.circleci.com/pipelines/github/esl/MongooseIM?branch=GH-actions-fix)

codecov/coveralls reports diff for 541b9c19 commit (before Jingle/SIP testing fix and with generated modules) is [here](https://docs.google.com/spreadsheets/d/1-FjJq6H_Kug57Rq3QusIL8qCBtSoelUe3S62mcDUO7g/edit#gid=1355518201)

TODO: some of the tests are flaky and this results in frequent GH actions jobs failures (especially MySQL jobs), this should be investigated and fixed in the following PRs.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/170851) / small_tests / 570ede1b1df79ae3b77f87efbbed307a6580417a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/170851/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/170851/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-21_13.57.45/lib.mongooseim.logs/run.2023-03-21_13.57.46/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/170913) / small_tests / 5d29461f4b3332ec8cda1392d06174ef720f0d3c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/170913/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/170913/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-21_20.44.03/lib.mongooseim.logs/run.2023-03-21_20.44.03/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/170941) / small_tests / a2c834c4b1b208669e73a36b4eb041c15391c9fd
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/170941/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/170941/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-21_21.19.10/lib.mongooseim.logs/run.2023-03-21_21.19.11/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171033) / small_tests / 49f2b684ff92782c1ea24055832b00f16c0bb3fd
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171033/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171033/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-22_10.32.52/lib.mongooseim.logs/run.2023-03-22_10.32.52/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171068) / small_tests / 9bdd3924133c9d04bb92fc708f1a66e9f7bc66bb
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171068/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171068/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-22_14.27.06/lib.mongooseim.logs/run.2023-03-22_14.27.06/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171092) / small_tests / 2d6f4046ec3b55fdb856eeb6ac6e5c019cbec920
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171092/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171092/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-23_01.40.26/lib.mongooseim.logs/run.2023-03-23_01.40.27/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171121) / small_tests / 068194f69aa4b33a94611770eb6cf4963f30fa7a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171121/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171121/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-23_01.43.36/lib.mongooseim.logs/run.2023-03-23_01.43.37/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171144) / small_tests / bdf6f172e7992039c38d0d9a03e87cdd8159706c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171144/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171144/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-23_01.49.02/lib.mongooseim.logs/run.2023-03-23_01.49.03/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171172) / small_tests / 0984deab3b2ce5cbdb613614f4c789aafd5168fd
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171172/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171172/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-23_01.53.50/lib.mongooseim.logs/run.2023-03-23_01.53.50/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171197) / small_tests / 09d93ddb9f9df50769db28d51b9b249ce1a47199
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171197/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171197/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-23_01.57.33/lib.mongooseim.logs/run.2023-03-23_01.57.33/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171226) / small_tests / b8bd1133969caaecba4840b633a0754a88553313
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171226/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171226/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-23_01.59.58/lib.mongooseim.logs/run.2023-03-23_01.59.59/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171288) / small_tests / ea8704ca128e6961cd96cce0931329ab2924d901
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171288/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171288/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-23_14.55.03/lib.mongooseim.logs/run.2023-03-23_14.55.03/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171341) / small_tests / b4fa2e11f23a16f046c168484d3115f68cd222b2
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171341/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171341/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-23_16.41.08/lib.mongooseim.logs/run.2023-03-23_16.41.08/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171421) / small_tests / b6f8c9d048d69fa090fca042bde254a0bf3b3bba
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171421/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171421/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-24_01.28.32/lib.mongooseim.logs/run.2023-03-24_01.28.32/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171448) / small_tests / 541b9c196ae798d005b0c16399f49d90bf92528a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171448/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171448/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-24_01.37.52/lib.mongooseim.logs/run.2023-03-24_01.37.52/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171511) / small_tests / f1b72372be0eae5d747dfd47389e4e9abe604ce0
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171511/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171511/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-24_15.32.32/lib.mongooseim.logs/run.2023-03-24_15.32.33/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171537) / small_tests / c2eddef0c2aaa3860872c043a0e6a1a35a5eb529
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171537/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171537/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-27_21.38.03/lib.mongooseim.logs/run.2023-03-27_21.38.03/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171564) / small_tests / 4169338c7444f26c1755f4a7ff20478431124baf
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171564/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171564/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-27_23.13.45/lib.mongooseim.logs/run.2023-03-27_23.13.45/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Looks good :ok_hand:

File: rebar.config [hunk 1]
```diff
@@ -185,7 +185,7 @@
  [
   {pc, "1.14.0"},
   {provider_asn1, "0.3.0"},
-  {rebar3_codecov, "0.3.0"},
```
[the remark this answers is not in the document]
Comment (DenysGonchar) [reply rebar.config:188]: Yes, we should. I was planning to discuss with @NelsonVides creation of the new hex release for rebar3_codecov plugin.

[the remark this answers is not in the document]
Comment (DenysGonchar) [reply rebar.config:188]: that was fast

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/171625) / small_tests / 474757b5c6b5fad461831e3bbce2c07edb581ff8
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3992/171625/small_tests.24.3.4.9-amd64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3992/171625/small_tests.24.3.4.9-amd64/small/ct_run.mongooseim@localhost.2023-03-28_10.29.35/lib.mongooseim.logs/run.2023-03-28_10.29.35/suite.log.html)

**Pull request merged**